mailbox/api/src/main/java/org/apache/james/mailbox/MailboxManager.java

```diff
@@ -273,24 +273,46 @@
 
     /**
      * Autenticates the given user against the given password.<br>
-     * When authentic and authorized, a session will be supplied
+     * When authenticated and authorized, a session will be supplied
      * 
      * @param userid
      *            user name
      * @param passwd
      *            password supplied
      * @param log
      *            context sensitive log
-     * @return a <code>MailboxSession</code> when the user is authentic and
+     * @return a <code>MailboxSession</code> when the user is authenticated and
      *         authorized to access
      * @throws BadCredentialsException
-     *             when system access is denighed for the given user
+     *             when system access is denied for the given user
      * @throws MailboxException
      *             when the creation fails for other reasons
      */
     MailboxSession login(String userid, String passwd, Logger log) throws BadCredentialsException, MailboxException;
 
     /**
+     * Autenticates the given administrator against the given password,
+     * then switch to the real user<br>
+     * When authenticated and authorized, a session for the real user will be supplied
+     * 
+     * @param adminUserId
+     *            user name of the admin user, matching the credentials
+     * @param passwd
+     *            password supplied for the admin user
+     * @param realUserId
```

Comment:
changed to other user as the naming of the function doesn't seem to be subject to debate